perceval/backends/core/gitter.py

```diff
@@ -0,0 +1,325 @@
+        kwargs = {
+            'from_date': from_date,
+        }
+
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch the messages
+
+        :param category: the category of items to fetch
+        :param kwargs: backend arguments
+
+        :returns: a generator of items
+        """
+        from_date = kwargs['from_date']
+        logger.debug("Get Gitter message paginated items of room: %s from date: %s", self.room, from_date)
+        from_date = str_to_datetime(from_date).timestamp()
+        fetching = True
+        before_id = None
+        num_msgs = 0
+        self.room_id = self.client.get_room_id()
+        logger.debug("Room Id of room: %s is %s", self.room, self.room_id)
+        page = 0
+        while fetching:
+            page += 1
+            logger.debug("Page: %i" % page)
+            message_group = self.client.message_page(self.room_id, before_id)
+
+            if not message_group:
+                fetching = False
+                continue
+
+            message_group = message_group.json()
+            for raw_message in message_group:
+
+                if str_to_datetime(raw_message['sent']).timestamp() > from_date:
+                    num_msgs += 1
+                    yield raw_message
+                else:
+                    fetching = False
+
+            before_id = message_group[0]['id']
+
+        logger.debug("Fetch process completed: %s message fetched", num_msgs)
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not support items resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Gitter item."""
+
+        return item['id']
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts and coverts the sent time of a message
+        from a Gitter item.
+
+        The timestamp is extracted from 'sent' field and
+        converted to a UNIX timestamp.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = str_to_datetime(item['sent'])
+
+        return ts.timestamp()
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Gitter item.
+
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        room = urijoin(self.group, self.room)
+
+        return GitterClient(self.api_token, self.max_items, self.archive,
```

Comment:
Hi @valeriocos. The only reason I have used roomid to init the client is because, I thought it would be a good idea to include the roomid in the search fields. Since roomid needs a separate API call as is not returned by the chatmessages API.
Let me know if it's not necessary? Or if you prefer any other implementation

Reply:
Sorry for not being clear @animeshk08 . The roomid in the search fields is a really good idea, but this is at the level of the backend. 

I was suggesting to remove the param `room` at https://github.com/chaoss/grimoirelab-perceval/pull/632/files#diff-6459a75c74db880450cc119cd49e3051R235. The info to retrieve the roomid can be passed as params of the method `get_room_id` (https://github.com/chaoss/grimoirelab-perceval/pull/632/files#diff-6459a75c74db880450cc119cd49e3051R278)

Reply:
Thank you for the clarification. I will make the change :)